test/powershell/engine/Module/ModulePath.Tests.ps1

```diff
@@ -187,4 +187,11 @@ Describe "SxS Module Path Basic Tests" -tags "CI" {
             Remove-Item -Path $userConfigPath -Force
         }
     }
+
+    It 'User PSModulePath has trailing separator' -Skip:(!$IsWindows) {
```

Comment:
What's the reason this only works on Windows?

Reply:
Only Windows has pre-defined `PSModulePath`.

Reply:
There are scenarios where it's pre-defined on *nix:

- Nested PowerShell process
- Environment variable set before startup

Reply:
I mean the test :-)

Reply:
Those both seem like scenarios we could test without too much trouble. Certainly (2) is just a case of setting the variable before the invocation and unsetting it afterward

Reply:
The real world scenario where this happens is only if the env var in the registry has a trailing slash.  In other cases, I couldn't get the problem to repro which is why I had a hard time adding a test case that failed without the fix.  This test emulates that behavior by adding the slash to the user path and removing the shared path so that it gets inserted.

Reply:
I think I can update the test to be cross platform